allennlp/common/tqdm.py

```diff
@@ -22,29 +28,32 @@
 _tqdm.monitor_interval = 0
 
 
-class Tqdm:
-    # These defaults are the same as the argument defaults in tqdm.
-    default_mininterval: float = 0.1
+logger = logging.getLogger("tqdm")
+logger.propagate = False
 
-    @staticmethod
-    def set_default_mininterval(value: float) -> None:
-        Tqdm.default_mininterval = value
 
-    @staticmethod
-    def set_slower_interval(use_slower_interval: bool) -> None:
-        """
-        If `use_slower_interval` is `True`, we will dramatically slow down `tqdm's` default
-        output rate.  `tqdm's` default output rate is great for interactively watching progress,
-        but it is not great for log files.  You might want to set this if you are primarily going
-        to be looking at output through log files, not the terminal.
-        """
-        if use_slower_interval:
-            Tqdm.default_mininterval = 10.0
-        else:
-            Tqdm.default_mininterval = 0.1
+class TqdmToLogsWriter(object):
+    def __init__(self):
+        self.last_message_written_time = 0
 
+    def write(self, message):
+        sys.stderr.write(message)
+        now = time()
+        if now - self.last_message_written_time >= 10 or "100%" in message:
```

Comment:
The one weirdness with this "100%" hack is that `tqdm` will often write two messages that say "100%" in rapid succession. I don't know why this happens. In principle, we want the second one of the two, because that's what you see on the console, and that's the one with the accurate items-per-second count. I can filter so it shows only the first one of the two, but not so that it only shows the second one. So I decided it's better to have the message there twice than to have an inaccurate items-per-second number in the logs.

Reply:
Could you add this as a comment here?